### ISSUE TITLE
cuelang.org: preprocessor doesn't reformat .java sources

### ISSUE THREAD

**Issue by jpluscplusm** [cue-lang/cue#3363]
### What page were you looking at?

https://cl-1198969-5--cue-cls.netlify.app/docs/tutorial/validate-data-schema-java/#step-6

### What version of the site were you looking at?

https://github.com/cue-lang/cuelang.org/commit/210b2235881bd2a634d4d8dd2a4d268942292e83

### What did you do?

I authored a page which contained a `.java` file that had inconsistent formatting. (See the occasional 8-space indent on the linked page, mixed in with 4-space-indent lines)

### What did you expect?

That the preprocessor would reformat `.java` source files (similarly to how it handles `.go` and `.cue` files) and remove the formatting burden from the author.

### What did you see instead?

Java sources are not reformatted.

**Comment by jpluscplusm** [comment]
Possible options: https://github.com/jhipster/prettier-java appears to be a plugin for https://prettier.io/ for Java source files.

**Comment by jpluscplusm** [comment]
More options found at https://github.com/rishirdua/awesome-code-formatters?tab=readme-ov-file#java:
- https://github.com/google/google-java-format 
- http://notzippy.github.io/JALOPY2-MAIN/

Also some options at https://github.com/rishirdua/awesome-code-formatters?tab=readme-ov-file#general-purpose

**Comment by mvdan** [comment]
I don't believe this is planned, and there's a mixture of languages that we might want to cover in terms of libcue integrations (C, Java, Python, Rust), where integrating with external formatters could get quite tricky. So I think it's best that we close this for now until there's a plan to do this relatively consistently, as opposed to just for Java.
